src/main.rs

```diff
@@ -475,3 +475,38 @@ fn all_versions(crd: &CustomResourceDefinition) -> String {
     vers.sort_by_cached_key(|v| std::cmp::Reverse(Version::parse(v).priority()));
     vers.join(", ")
 }
+
+fn format_docstr(indent: &str, input: &str) -> String {
+    let rustdoc_code = "```\n";
+    let rustdoc_starts: Vec<usize> = input.match_indices(rustdoc_code).step_by(2).map(|(index, _)| index).collect();
```

Comment:
Maybe a comment here about how this is a best-effort attempt to try to match opening and closing backticks.

I have a feeling that this has several places it can go wrong:
- trailing spaces before newline
- uneven number of triple-backtick pairs (invalid markdown)
- a triple backtick starter that already contains a language `'''go` (which might lead to uneven matches?)

i suspect that while a lot of this would hopefully be caught by linters, maybe a regex here that makes some of this optional (stray whitespace, optional languages) can make this a little safer? WDYT?

Reply:
Thanks for the input... Didn't think of the language one, nor trailing white spaces tbh... Ignored the invalid MD tho.
I was trying to avoid to impl. a "full-blown code snippet" scanner for MD. 

Are you thinking of leveraging something like [this](https://docs.rs/regex/latest/regex/struct.Regex.html#method.captures_iter) instead?

Reply:
That's fair. I'm just hoping not to have to debug some awkward CRD in the future that does this wrong.

Yeah, that link there is was what I was thinking (with a regex on a pair of triples with arbitrary contents). At least then either the match works, or it does nothing, which feels safer. Might be a bit hairier to write though...

Reply:
Have a look at the changes in 99ec9079dfa03bfebc96d5777dccd6622601b58f
Your comment:

>I'm just hoping not to have to debug some awkward CRD in the future that does this wrong.

made me think maybe _not_ being smart is the way to go here. I am _not_ removing what ever follows the triple back ticks, but put it on a new line, wdyt? I hope this actually helps understand what's going on. All remains in a `text` block tho

Reply:
That does look safer in some ways, but it's also pretty complicated now; multiple instances of array indexing, array slicing, byte conversion, ascii checks, 3 mutable variables in a loop.

Maybe I am wrong, but thought a regex here with [replace_all](https://docs.rs/regex/latest/regex/struct.Regex.html#method.replace_all) could do this in one call.

Something like; https://github.com/jonboh/obsidian2neorg/blob/51ac21da0854daed8974b7e64bb3c5d08e570e9a/src/main.rs#L42-L46

Reply:
Yeah, I can see how that's a little more difficult for humans to parse. Arguably, regex are just as difficult & complex (tho the complexity gets hidden away)… Alright, I'll give it a shot to replace it all with a regex! 👌